DNN Platform/Library/Services/Mail/SendTokenizedBulkEmail.cs

```diff
@@ -347,24 +347,22 @@ private List<Attachment> LoadAttachments()
 			foreach (var attachment in _attachments)
 			{
 				var buffer = new byte[4096];
-                using (var memoryStream = new MemoryStream())
+                var memoryStream = new MemoryStream();
```

Comment:
I would prefer if there's some defensive code to ensure the stream gets disposed if there's an exception loading the data into it.  Something like this:

```
Attachment newAttachment;
MemoryStream memoryStream = null;
try
{
    memoryStream = new MemoryStream();
	// the while loop
    newAttachment = new Attachment(memoryStream, attachment.ContentType);
    memoryStream = null;
}
finally 
{
	memoryStream?.Dispose();
}
```

Reply:
@bdukes `try/finally` is the same as `using` when you dispose in the `finally`.  This will not work.  Did you mean a `try/catch` perhaps?

Reply:
@david-poindexter this pattern only disposes if there's an exception thrown.  After the `Attachment` is constructed, `memoryStream` is set to `null`, so that in the `fianlly` block nothing is disposed.

Reply:
The "LoadAttachments" method effectively copies the attachments of the email into new streams. This is then used in the email method call to actually send a bulk email. There is a potential for a lot of these copies being made. So there is a case to be made to properly clean this up. But IMHO that needs to be done by the code that sends the email. IMO the issue requires a bigger intervention in this class.

Reply:
@bdukes okay, now I see what you are saying.  I will test that pattern and push a new commit if it works.  Thanks!

Reply:
@donker are you in agreement with the suggested pattern made by @bdukes?  Or are you saying you believe this will be problematic too?  I have not had the ability to test this yet due to some serious computer issues I am dealing with.  Hopefully, I'll be back up and running soon.

Reply:
I would agree with @bdukes on this one, we need at least some form of protection, otherwise, we will have a memory leak situation and given that things aren't exactly perfect in this area, I think it is a requirement.

I also agree with @donker that longer-term this should most likely be done in a more efficient manner 

Looking at the C# reference documentation for SMTP and email sending, there is a best practices recommendation for Using statement (Which would be expected)

Reply:
@bdukes and @mitchelsellers - just pushed changes, worked like a charm.  Thanks again for the review and recommendations for this fix.  In the bigger picture, this area needs some more TLC.